lib/eth/abi/type.rb

```diff
@@ -98,15 +109,13 @@ def ==(another_type)
       def size
         s = nil
         if dimensions.empty?
-          unless ["string", "bytes"].include?(base_type) and sub_type.empty?
+          if !(["string", "bytes", "tuple"].include?(base_type) and sub_type.empty?)
```

Comment:
+1, but can you change the `and` to `&&`?

`and` should not be used for conditionals (or pretty much ever) due to its lower operator precedence. It can cause unexpected bugs.